src/mongoModelsRepo/createPersona.ts

```diff
@@ -8,20 +8,20 @@ import { PERSONAS_COLLECTION } from './utils/constants/collections';
 export default (config: Config) => {
   return async (opts: CreatePersonaOptions): Promise<CreatePersonaResult> => {
     const collection = (await config.db).collection(PERSONAS_COLLECTION);
-
-    const docToInsert = {
-      name: opts.name,
-      organisation: new ObjectID(opts.organisation),
-    };
+    const personaId = new ObjectID();
 
     // Docs: http://mongodb.github.io/node-mongodb-native/2.2/api/Collection.html#insertOne
     // Docs: http://bit.ly/insertOneWriteOpResult
-    const opResult = await collection.insertOne(docToInsert, {});
+    const opResult = await collection.insertOne({
+      _id: personaId,
+      name: Boolean(opts.name) ? opts.name : personaId.toString(),
+      organisation: new ObjectID(opts.organisation),
+    }, {});
 
     // Formats the persona to be returned.
     const persona: Persona = {
       id: opResult.insertedId.toString(),
-      name: opts.name,
+      name: Boolean(opts.name) ? opts.name : personaId.toString(),
```

Comment:
In future, we should try to use a variable instead to avoid code duplication (line 24 and line 17).